get_data.py

```diff
@@ -0,0 +1,103 @@
+"""Module to Store Data From the CSV files into data Structures and Handle the Parsed Commands"""
+
+import os
+import csv
+from daily_weather_data import WeatherDataRow
+
+
+class WeatherData:
+    """
+    Class containing 1 class variable (list) to use  for months of the year
+    Contains attributes for file path , year to display and the month for each object
+    Contains a method to load monthly and yearly data
+    Contains an internal function which cuts out useless columns from the Monthly_weather dictionary and yearly_weather dictionaries
+    """
+
+    months_of_the_year = [
+        "jan",
+        "feb",
+        "mar",
+        "apr",
+        "may",
+        "jun",
+        "jul",
+        "aug",
+        "sep",
+        "oct",
+        "nov",
+        "dec",
+    ]
+
+    def __init__(self, parser_file_path, year, month):
+        """initialize function to store attributes affter instantiation of a class"""
+
+        self.parser_file_path = parser_file_path
+        self.year = year
+        self.month = month
+
+    def load_data(self, month):
+        """
+        Load data for a specific month or all months if month is None.
+        Returns a List of Objects if the month is given in the format :
+
+        list = [obj1 , obj2 , obj3]  each obj contains values for Max , Min temp and humidity
+
+        The Yearly Data consists of a 2D list conttaining 12 Lists of Objects ,corresponding to each month
+        with each object corresponding to a single day of the month
+
+        2Dlist_yearly = [[obj1 , obj2 , obj3] , [obj4 , obj5 , obj6]]
+        """
+        total_data = []
+        if month is not None:
+            # Load data for a specific month
+            if month < 1 or month > 12:
+                raise ValueError("Invalid month value. Month must be between 1 and 12.")
+            file_path = os.path.join(
+                self.parser_file_path,
+                f"Murree_weather_{self.year}_{WeatherData.months_of_the_year[month - 1]}.txt",
+            )
+            total_data.extend(self.load_data_from_file(file_path))
+            return total_data
+        else:
+            # Load data for all months in the year
+            for month_index in range(1, 13):
+                file_path = os.path.join(
+                    self.parser_file_path,
+                    f"Murree_weather_{self.year}_{WeatherData.months_of_the_year[month_index - 1]}.txt",
+                )
+                try:
+                    total_data.extend(self.load_data_from_file(file_path))
+
+                except FileNotFoundError as fe:
+                    continue
+            return total_data
```

Comment:
This function can be further simplified. Think of a way to combine the code snippets inside `if-else` statement.